metricbeat/module/kubernetes/state_container/state_container.go

```diff
@@ -138,9 +138,20 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 	m.enricher.Enrich(events)
 
 	for _, event := range events {
+		containerFields := mapstr.M{}
+
+		// Add container.name field (same as kubernetes.container.name)
+		if containerName, ok := event["name"]; ok {
+			cName, ok := (containerName).(string)
+			if !ok {
+				m.Logger().Debugf("Error while casting containerImage: %s", ok)
+			} else {
+				kubernetes.ShouldPut(containerFields, "name", cName, m.Logger())
+			}
+		}
+
 		// applying ECS to kubernetes.container.id in the form <container.runtime>://<container.id>
 		// copy to ECS fields the kubernetes.container.image, kubernetes.container.name
```

Comment:
I think this comment is also applicable to the `container.name`

Reply:
I checked, but the value of the field is just `<container.name>`, it does not have the same format as `<container.runtime>://<container.id>`.

Reply:
Thanks for the PR @constanca-m ! I have tested this locally and works fine. 

(I am posting the example from my local setup:)

metricset: container -> container.name: nginx
metricset: state_container: -> container.name: nginx

Even works for mutiple containers inside same pod

Reply:
I think what @tetianakravchenko meant is that the comment `copy to ECS fields the kubernetes.container.image, kubernetes.container.name` applies also for the addition you did. So it would be better to move your code under this comment.